File: tugas-fp/basket/basket.py
# This and the cpp one would TLE for large numbers.
n = int(input())
inputs = [list(map(int, input().split())) for _ in range(n)]

for i in inputs:
    positions = list(range(1, i[0] + 1))
    target = i[1]
    rest = []
    layup = False

    while len(rest) < target:
        print(rest)
        head = positions.pop(0)
        if not layup:
            rest.append(head)
        else:
            positions.append(head)
        layup = not layup

    print(rest[-1])

File: tugas-fp/basket/basket.cpp
#include <iostream>
#include <vector>

using namespace std;

int main()
{
    vector<pair<int, int>> inputs;
    int tc;
    cin >> tc;

    for (int i = 0; i < tc; i++)
    {
        int temp[2];
        cin >> temp[0] >> temp[1];
        inputs.push_back(make_pair(temp[0], temp[1]));
    }

    for (int i = 0; i < tc; i++)
    {
        vector<int> range;
        for (int j = 1; j <= inputs[i].first; j++)
        {
            range.push_back(j);
        }

        bool layup = false;
        vector<int> rest;
        int ll = 2 * range.size();

        for (int j = 0; j < ll && rest.size() < inputs[i].second; j++)
        {
            int head = range.front();
            range.erase(range.begin());

            if (layup)
            {
                range.push_back(head);
                layup = false;
            }
            else
            {
                rest.push_back(head);
                layup = true;
            }
        }

        cout << rest.back() << endl;
    }
}